cmdstanpy/stanfit/mcmc.py

```diff
@@ -78,31 +77,51 @@ def __init__(
         assert isinstance(
             sampler_args, SamplerArgs
         )  # make the typechecker happy
-        iter_sampling = sampler_args.iter_sampling
-        if iter_sampling is None:
-            self._iter_sampling = _CMDSTAN_SAMPLING
-        else:
-            self._iter_sampling = iter_sampling
-        iter_warmup = sampler_args.iter_warmup
-        if iter_warmup is None:
-            self._iter_warmup = _CMDSTAN_WARMUP
-        else:
-            self._iter_warmup = iter_warmup
-        thin = sampler_args.thin
-        if thin is None:
-            self._thin: int = _CMDSTAN_THIN
-        else:
-            self._thin = thin
+        self._iter_sampling:int = _CMDSTAN_SAMPLING
+        if sampler_args.iter_sampling is not None:
+            self._iter_sampling = sampler_args.iter_sampling
+        self._iter_warmup:int = _CMDSTAN_WARMUP
+        if sampler_args.iter_warmup is not None:
+            self._iter_warmup = sampler_args.iter_warmup
+        self._thin:int = _CMDSTAN_THIN
+        if sampler_args.thin is not None:
+            self._thin = sampler_args.thin
         self._is_fixed_param = sampler_args.fixed_param
         self._save_warmup = sampler_args.save_warmup
         self._sig_figs = runset._args.sig_figs
+
         # info from CSV values, instantiated lazily
+        self._draws: np.ndarray = np.array(())
+        # only valid when not is_fixed_param
         self._metric: np.ndarray = np.array(())
         self._step_size: np.ndarray = np.array(())
-        self._draws: np.ndarray = np.array(())
+        self._divergences: np.ndarray = np.zeros(self.runset.chains, int)
+        self._max_treedepths: np.ndarray = np.zeros(self.runset.chains, int)
+
         # info from CSV initial comments and header
         config = self._validate_csv_files()
         self._metadata: InferenceMetadata = InferenceMetadata(config)
+        # prelim diagnostics
```

Comment:
How would you feel about putting this in a function? something like `self._report_diagonostics()`